### PR TITLE
[MGDOBR-361] Split the scenarios into many features

### PR DESCRIPTION
**Pull request by radtriste**
[MGDOBR-361](https://issues.redhat.com/browse/MGDOBR-361) 

- Refactor the feature files
- Updated `run-local-tests` script to run in parallel and/or with tags
- Added some authentication scenarios to bridge/processor functionalities
- Separated the Ingress full scenario into separate ones (json, plain and wrong endpoint)

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

**Review by jiripetrlik — APPROVED** [review]
Looks good to me. One minor comment.

File: shard-operator-integration-tests/run-local-tests.sh
Comment (radtriste) [reply shard-operator-integration-tests/run-local-tests.sh:1]: yeah, just thought to copy this script to this directory as it can be useful for anybody ...

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java:70]: Nitpick - the headers are required just for plain request, when cloud event request is provided then the values are taken from request body and no header is required. Though that this would mean a need to specify headers as part of test step.

**Comment by sutaakar** [comment]
Looks good to me.
For the Processor definitions - you can remove filters as they are not needed for basic testing.
For the cloud events, you can simplify data part, one entry could be enough.

That can reduce the scenario text size while keeping all the needed functionality.

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java
Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java:70]: I can differentiate `json` and `plain` cloud event

Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/IngressSteps.java:70]: added the `and default headers` as a possible steps for `send cloud event`

**Comment by radtriste** [comment]
@sutaakar I applied your comments. Let me know :)

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java
Comment (jomarko) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java:51]:
```suggestion
                    "Cannot create and initiate a random bridge name correctly. Please cleanup the environment...");
```

Comment (jomarko) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/BridgeSteps.java:51]: small typo?

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/ProcessorSteps.java
Comment (jomarko) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/ProcessorSteps.java:79]: Could we please close this stream once not needed? I believe using try with resources could solve it easily.

**Review by jomarko — COMMENTED** [review]
Thank you for PR @radtriste some minor comments from me inline.

Also one general question, probably for anyone. We check negative scenarios - failing http requests with specific error codes 401, 404. Thinking if we can not somehow simplify the cucumber steps, for these "failing response scenarios" . For example do just a check code != 200 or something similar?

**Comment by sutaakar** [comment]
@jomarko From my perspective these response codes are important. If we would check something like != 200 then we may miss possible errors like some manager exception with returning 500 instead of proper 4xx.
Also I see these tests as sort of a documentation describing behavior for different scenarios. That way we can see what is returned for what case.

File: integration-tests/src/test/resources/features/processor.feature
Comment (sutaakar) [inline integration-tests/src/test/resources/features/processor.feature:116]: Can you please add a scenario with deleting non existing processor? It should be fixed by https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/396

File: integration-tests/src/test/resources/features/bridge.feature
Comment (sutaakar) [inline integration-tests/src/test/resources/features/bridge.feature:33]: Can you please add a test deleting non existing bridge?

**Comment by sutaakar** [comment]
Otherwise it looks good to me

File: integration-tests/src/test/resources/features/bridge.feature
Comment (radtriste) [reply integration-tests/src/test/resources/features/bridge.feature:33]: added. See https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/394/commits/895705998957de1b5329aff1430855cdfa09648e

File: integration-tests/src/test/resources/features/processor.feature
Comment (radtriste) [reply integration-tests/src/test/resources/features/processor.feature:126]: I get a 500 ...

Comment (radtriste) [reply integration-tests/src/test/resources/features/processor.feature:126]: logs: 
```
Request method:	DELETE
Request URI:	http://192.168.49.2:80/manager/api/v1/bridges/ff79166a-cf6f-4269-9a9e-2f8a2c6c01f6/processors/myProcessor-3239
Proxy:			<none>
Request params:	<none>
Query params:	<none>
Form params:	<none>
Path params:	<none>
Headers:		Accept=*/*
				Content-Type=application/json
Cookies:		<none>
Multiparts:		<none>
Body:			<none>

HTTP/1.1 500 Internal Server Error
Date: Fri, 04 Mar 2022 14:52:56 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 72
Connection: keep-alive

{
    "details": "Error id 9853e7ea-b1eb-4aa8-ae86-af62a1f3684c-3",
    "stack": ""
}
```
```
2022-03-04 14:52:56,404 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-11) HTTP Request to /api/v1/bridges/ff79166a-cf6f-4269-9a9e-2f8a2c6c01f6/processors/myProcessor-3239 failed, error id: 9853e7ea-b1eb-4aa8-ae86-af62a1f3684c-3: org.jboss.resteasy.spi.UnhandledException: java.lang.NullPointerException
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:105)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:359)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:91)
	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NullPointerException
	at com.redhat.service.bridge.manager.ProcessorServiceImpl.updateProcessorConnectorForDeletionOnDB(ProcessorServiceImpl.java:189)
	at com.redhat.service.bridge.manager.ProcessorServiceImpl_Subclass.updateProcessorConnectorForDeletionOnDB$$superforward1(Unknown Source)
	at com.redhat.service.bridge.manager.ProcessorServiceImpl_Subclass$$function$$5.apply(Unknown Source)
	at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)

```

Comment (radtriste) [reply integration-tests/src/test/resources/features/processor.feature:126]: or maybe I don't have the latest version. gonna check that

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/ProcessorSteps.java
Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/ProcessorSteps.java]: updated

Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/steps/ProcessorSteps.java]: See https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/394/commits/f232b5a4c4021862eebe9af536beed3ddabf6fe0

File: integration-tests/src/test/resources/features/processor.feature
Comment (sutaakar) [reply integration-tests/src/test/resources/features/processor.feature:126]: probably

**Review by sutaakar — APPROVED** [review]

File: integration-tests/src/test/resources/features/processor.feature
Comment (radtriste) [reply integration-tests/src/test/resources/features/processor.feature:126]: verified OK with latest version :)

**Comment by radtriste** [comment]
@jomarko waiting for your approval and then it can be merged

**Review by jomarko — APPROVED** [review]
Thank you @radtriste

**Review by Srihari1192 — APPROVED** [review]

**Comment by sutaakar** [comment]
@radtriste Can you revert the debug changes and rerun tests? It should be more stable now.

**Pull request merged**
